beacon-chain/rpc/beaconv1/blocks.go

```diff
@@ -24,20 +118,235 @@ func (bs *Server) ListBlockHeaders(ctx context.Context, req *ethpb.BlockHeadersR
 // new block into its state, and therefore validate the block internally, however blocks which fail the validation are
 // still broadcast but a different status code is returned (202).
 func (bs *Server) SubmitBlock(ctx context.Context, req *ethpb.BeaconBlockContainer) (*ptypes.Empty, error) {
-	return nil, errors.New("unimplemented")
+	blk := req.Message
+	root, err := blk.HashTreeRoot()
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Could not tree hash block: %v", err)
+	}
+
+	v1alpha1Block, err := v1ToV1alpha1Block(&ethpb.SignedBeaconBlock{Block: blk, Signature: req.Signature})
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Could not convert block to v1")
+	}
+
+	// Do not block proposal critical path with debug logging or block feed updates.
+	defer func() {
+		log.WithField("blockRoot", fmt.Sprintf("%#x", bytesutil.Trunc(root[:]))).Debugf(
+			"Block proposal received via RPC")
```

Comment:
Is this necessary? In this method, you have two debug messages which are essentially the same.